src/saltext/vmware/modules/esxi.py

```diff
@@ -4179,13 +4189,21 @@ def check_compliance(profile=None, cluster_paths=None, desired_state_spec=None,
     """
     
     log.info("Checking complaince %s", desired_state_spec)
+    desired_state_spec = convert_ordered_dict_to_dict(desired_state_spec)    
     config = __opts__
-    if desired_state_spec == None:
-        desired_state_spec = json.loads(json.dumps(desired_state_spec))
     if not esx_config:
         esx_config = utils_esxi.create_esx_config(config, profile)
-    response_check_compliance = esx_config.check_compliance(desired_state_spec= desired_state_spec, cluster_paths= cluster_paths)
-    return response_check_compliance
+  
+	try:
```

Comment:
pre-commit probably failing since indentation is incorrect
And do you really need to convert and ordereddict to a regular dict, is there no simpler python way to do that ?